src/code/templates/fv-genome-challenge.js

```diff
@@ -0,0 +1,218 @@
+
+/*
+ * HatchDrakeButton
+ */
+const HatchDrakeButton = ({label, onClick, disabled}) => {
+
+  function handleClick() {
+    onClick();
+  }
+
+  return (
+    <div className={classNames('hatch-drake-button', {disabled})} onClick={handleClick}>
+      <div className="button-label hatch-drake-button-label unselectable">
+        {label}
+      </div>
+    </div>
+  );
+};
+
+HatchDrakeButton.propTypes = {
+  label: PropTypes.string.isRequired,
+  onClick: PropTypes.func.isRequired,
+  disabled: PropTypes.bool
+};
+
+/*
+ * YourDrakeView
+ */
+class YourDrakeView extends React.Component {
+
+  static propTypes = {
+    org: PropTypes.object,
+    className: PropTypes.string,
+    hatchStarted: PropTypes.bool,
+    hatchComplete: PropTypes.bool,
+    onHatchComplete: PropTypes.func
+  }
+
+  render() {
+    const { org, className, hatchStarted, hatchComplete, onHatchComplete } = this.props;
+    return (
+      <div className='your-drake-surround'>
+        <EggHatchDrakeView drake={org} id="your-drake" className={className}
+                            hatchStarted={hatchStarted} hatchComplete={hatchComplete}
+                            onHatchComplete={onHatchComplete}
+                            width={300} />
+      </div>
+    );
+  }
+}
+
+/*
+ * TargetDrakeView
+ */
+class TargetDrakeView extends React.Component {
+
+  static propTypes = {
+    showIntro: PropTypes.bool,
+    org: PropTypes.object
+  }
+
+  state = {
+    introComplete: false
+  }
+
+  handleIntroComplete = () => {
+    this.setState({ introComplete: true });
+  }
+
+  render() {
+    const { showIntro, org } = this.props,
+          { introComplete } = this.state,
+          completeDisplayStyle = { display: showIntro && !introComplete ? 'none' : 'flex' },
+          introAnimation = <AnimatedSprite
+                              classNames='target-gizmo-animation'
+                              frames={5} frameWidth={915} duration={1600}
+                              onEnd={this.handleIntroComplete} />,
+          // instantiating with {display:none} allows images to preload during animation
+          completeView = <div className='target-drake-gizmo' style={completeDisplayStyle}>
+                            <div id="target-drake-label">
+                              {t('~LABEL.TARGET_DRAKE')}
+                            </div>
+                            <OrganismView id="target-drake" org={org} width={300} />
+                          </div>;
+    return (
+      <div className='target-drake-view'>
+        { showIntro && !introComplete ? introAnimation : null }
+        { completeView }
+      </div>
+    );
+  }
+}
+
+/*
+ * FVGenomeChallenge
+ */
+export default class FVGenomeChallenge extends React.Component {
+
+  static backgroundClasses = 'fv-layout fv-layout-a'
+
+  state = {
+    hatchStarted: false
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (!this.props.showUserDrake && !this.props.userDrakeHidden && nextProps.userDrakeHidden)
```

Comment:
I realized this is another confusing piece- `userDrakeHidden` and `showUserDrake` sound like they should be opposites. In reality, `showUserDrake` is referring to whether we're on a level that shows the drake, where `userDrakeHidden` refers to the current state of a level where we don't show the user drake by default.